pyanaconda/modules/subscription/subscription.py

```diff
@@ -509,3 +515,41 @@ def rhsm_observer(self):
         :rtype: RHSMObserver instance
         """
         return self._rhsm_observer
+
+    @property
+    def rhsm_config_defaults(self):
+        """Return RHSM config default values.
+
+        We need to have these available in case the user decides
+        to return to default values from a custom value at
+        runtime.
+
+        NOTE: The property is lazy evaluated, the first attempt
+        to read it fetches the full config dict from RHSM and
+        subsequent reads are then served from cache. Due to this
+        it is important no to set RHSM configuration values
+        first "warming up" this property by reading it or else
+        it might return non-default (Anaconda overwritten) data.
+
+        :return : dictionary of default RHSM configuration values
+        :rtype: dict
+        """
+        if self._rhsm_config_defaults is None:
+            # config defaults cache not yet populated, do it now
+            proxy = self.rhsm_observer.get_proxy(RHSM_CONFIG)
+            self._rhsm_config_defaults = proxy.GetAll("")
```

Comment:
Could you please create a constant for the empty string? Otherwise, it is a little confusing. It can be done in another PR.

Reply:
Yeah, that's a good point. I'll do a separate PR for it as it's already used in other places.

BTW, about the locale string - its used to specify locale for the error messages RHSM returns, so we only set it to the actual locale when the error message is going to be shown to the user in the UI. In other cases we keep the default value so that any error messages that get logged are not translated.